cpp/src/arrow/array/array_list_test.cc

```diff
@@ -197,6 +197,37 @@ class TestListArray : public ::testing::Test {
     EXPECT_FALSE(left->Slice(offset)->Equals(right->Slice(offset)));
   }
 
+  void TestFromArraysWithNullBitMap() {
+    std::shared_ptr<Array> offsets_w_nulls, offsets_wo_nulls, values;
+
+    std::vector<offset_type> offsets = {0, 1, 1, 3, 4};
+
+    std::vector<bool> offsets_w_nulls_flags = {true, false, true, true, true};
```

Comment:
The convention in other tests is to call this `xxx_is_valid`:
```suggestion
    std::vector<bool> offsets_w_nulls_is_valid = {true, false, true, true, true};
```